test/integration/cases_test.go

```diff
@@ -35,6 +37,89 @@ import (
 	jobset "sigs.k8s.io/jobset/api/jobset/v1alpha2"
 )
 
+var _ = Describe("Nodepool metrics", func() {
+	Context("When reconciling a resource", func() {
+		ctx := context.Background()
+
+		jsRef := types.NamespacedName{
+			Name:      "test-jobset",
+			Namespace: "default",
+		}
+
+		jobRef := types.NamespacedName{
+			Name:      "test-job",
+			Namespace: "default",
+		}
+
+		nps, err := gkeClient.ListNodePools(ctx)
+		if err != nil {
+			Fail("Failed to list node pools: " + err.Error())
+		}
+		var np = nps[0]
+
+		var node = &corev1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-node",
+				Labels: map[string]string{
+					"cloud.google.com/gke-nodepool": "test-nodepool",
+				},
+			},
+		}
+
+		var pod = &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-pod",
+				Namespace: "default",
+				Labels: map[string]string{
+					"jobset.sigs.k8s.io/jobset-name":           jsRef.Name,
+					"batch.kubernetes.io/job-name":             jobRef.Name,
+					"batch.kubernetes.io/job-completion-index": "0",
+				},
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:  "test-container",
+						Image: "busybox",
+					},
+				},
+				RestartPolicy: corev1.RestartPolicyNever,
+				NodeName:      node.Name,
+			},
+		}
+
+		It("should watch a Node", func() {
+			Expect(k8sClient.Create(ctx, node)).To(Succeed())
+		})
+
+		// Necessary because pod reconciler uses a cached client
+		// which is eventually consistent w k8sClient here
+		time.Sleep(5 * time.Second)
+
+		It("should watch a Pod", func() {
+			Expect(k8sClient.Create(ctx, pod)).To(Succeed())
+		})
+
+		// Necessary because pod reconciler uses a cached client
+		// which is eventually consistent w k8sClient here
+		time.Sleep(5 * time.Second)
+
+		It("should publish nodepool metrics", func() {
+			nodepool := expectedMetricsForNodePool(np, jsRef.Name, jobRef.Name)
+			assertMetrics(
```

Comment:
Let's assert on the value of the `job_scheduled` and `up` metrics.

Reply:
`nodepool_down_time_seconds` has a different value each time, but added for the others